internal/util/projutil/project_util.go

```diff
@@ -215,8 +215,15 @@ func GetOperatorType() OperatorType {
 }
 
 func IsOperatorGo() bool {
-	_, err := os.Stat(mainFile)
-	return err == nil
+	_, errGoMod := os.Stat(goModFile)
+	_, errGopkgTOMLFile := os.Stat(gopkgTOMLFile)
+	_, errPkgApiDir:= os.Stat(pkgApiDir)
+
+	// check fif has go modules or dep files or pkg/api
+	if errGoMod == nil || errGopkgTOMLFile == nil || errPkgApiDir == nil {
+		return true
+	}
+	return false
```

Comment:
I suck at boolean logic but I'm wondering if this could get reduced to:

```
return errGoMod == nil || errGopkgTOMLFile == nil || errPkgApiDir == nil
```

Reply:
Definitely 👍